File: Android/app/src/androidTest/java/com/etiennebaudoux/clipboardzanager/mocks/CloudAuthenticationMock.java
package com.etiennebaudoux.clipboardzanager.mocks;

import com.etiennebaudoux.clipboardzanager.componentmodel.cloudstorage.AuthenticationResult;
import com.etiennebaudoux.clipboardzanager.componentmodel.cloudstorage.CloudAuthentication;
import com.etiennebaudoux.clipboardzanager.componentmodel.core.StringUtils;
import com.etiennebaudoux.clipboardzanager.componentmodel.core.tasks.Task;

/**
 * Created by ebaud on 2/11/2017.
 */

public class CloudAuthenticationMock implements CloudAuthentication {
    @Override
    public Task<AuthenticationResult> authenticateAsync(String authenticationUri, String redirectUri) {
        return new Task<AuthenticationResult>(() -> {
            if (StringUtils.isNullOrEmpty(authenticationUri))
            {
                return new AuthenticationResult(true, redirectUri);
            }

            return new AuthenticationResult(false, redirectUri);
        });
    }
}

File: ClipboardZanager/Tests/ClipboardZanager.Shared.Tests/Mocks/CloudAuthenticationMock.cs
using System;
using System.Threading.Tasks;
using ClipboardZanager.Shared.CloudStorage;

namespace ClipboardZanager.Shared.Tests.Mocks
{
    internal class CloudAuthenticationMock : ICloudAuthentication
    {
        public Task<AuthenticationResult> AuthenticateAsync(string authenticationUri, string redirectUri)
        {
            return Task.Run(() =>
            {
                if (string.IsNullOrEmpty(authenticationUri))
                {
                    return new AuthenticationResult(true, new Uri(redirectUri));
                }

                return new AuthenticationResult(false, new Uri(redirectUri));
            });
        }
    }
}
